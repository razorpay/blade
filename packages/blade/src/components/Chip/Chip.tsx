import React from 'react';
import { useChipGroupContext } from './ChipGroupContext';
import {
  chipIconSizes,
  chipTextSizes,
  chipColorTokens,
  getChipInputHoverTokens,
  chipHeightTokens,
  chipHorizontalPaddingTokens,
} from './chipTokens';
import type { ChipProps } from './types';
import { AnimatedChip } from './AnimatedChip';
import { StyledChipWrapper } from './StyledChipWrapper';
import isEmpty from '~utils/lodashButBetter/isEmpty';
import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
import { getStyledProps } from '~components/Box/styledProps';
import BaseBox from '~components/Box/BaseBox';
import { SelectorLabel } from '~components/Form/Selector/SelectorLabel';
import { SelectorInput } from '~components/Form/Selector/SelectorInput';
import type { BladeElementRef } from '~utils/types';
import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
import { throwBladeError } from '~utils/logger';
import { useCheckbox } from '~components/Checkbox/useCheckbox';
import { useRadio } from '~components/Radio/useRadio';
import { isReactNative, makeSize, useBreakpoint } from '~utils';
import { Text } from '~components/Typography';
import { useTheme } from '~components/BladeProvider';

type OnChange = ({
  isChecked,
  event,
  value,
}: {
  isChecked: boolean;
  event?: React.ChangeEvent;
  value?: string;
}) => void;

const _Chip: React.ForwardRefRenderFunction<BladeElementRef, ChipProps> = (
  { isDisabled, value, children, icon: Icon, intent, color, testID, ...styledProps },
  ref,
) => {
  const { theme } = useTheme();
  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
  const groupProps = useChipGroupContext();
  const isInsideGroup = !isEmpty(groupProps);
  const [isPressed, setIsPressed] = React.useState(false);

  if (__DEV__) {
    if (!isInsideGroup) {
      throwBladeError({
        moduleName: 'Chip',
        message:
          '<Chip /> component should only be used within the context of a <ChipGroup /> component',
      });
    }
  }

  const _isDisabled = isDisabled ?? groupProps?.isDisabled;
  const _name = groupProps?.name;
  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
  const _isChecked = groupProps?.state?.isChecked(value!);
  // Check if the defaultValue property of groupProps is undefined
  const defaultChecked =
    typeof groupProps?.defaultValue === 'undefined'
      ? undefined // If undefined, defaultChecked is also undefined
      : groupProps?.defaultValue?.includes(value as string); // If multiple selection, check if value is in defaultValue array
  const useChip = groupProps?.selectionType === 'single' ? useRadio : useCheckbox;
  const _size = groupProps?.size || 'small';
  const _intent = intent ?? groupProps?.intent;
  // If intent is proovided and it's not none, use intent otherwise use color
  const _color = color ?? groupProps?.color ?? 'default';
  const chipColor = _intent && _intent !== 'none' ? _intent : _color;

  const handleChange: OnChange = ({ isChecked, value }) => {
    if (isChecked) {
      groupProps?.state?.addValue(value!);
    } else {
      groupProps?.state?.removeValue(value!);
    }
  };

  const { state, inputProps } = useChip({
    defaultChecked,
    isChecked: _isChecked,
    isDisabled: _isDisabled,
    name: _name,
    value,
    onChange: handleChange,
  });

  const handlePointerPressedIn = React.useCallback(() => {
    if (_isDisabled) return;
    setIsPressed(true);
  }, [_isDisabled]);

  const handlePointerPressedOut = React.useCallback(() => {
    if (_isDisabled) return;
    setIsPressed(false);
  }, [_isDisabled]);

  const handleKeyboardPressedIn = React.useCallback(
    (e: React.KeyboardEvent) => {
      if (_isDisabled) return;
      if (e.key === ' ') {
        setIsPressed(true);
      }
    },
    [_isDisabled],
  );

  const handleKeyboardPressedOut = React.useCallback(
    (e: React.KeyboardEvent) => {
      if (_isDisabled) return;
      if (e.key === ' ') {
        setIsPressed(false);
      }
    },
    [_isDisabled],
  );

  let textVariant = 'unchecked';
  if (_isChecked && chipColor) {
    textVariant = chipColor;
  }
  if (_isDisabled) {
    textVariant = 'disabled';
  }
  const chipTextColor = chipColorTokens.text[textVariant];
  const chipIconColor = chipColorTokens.icon[textVariant];

  let intentVariant = 'unchecked';
  const stateVariant = _isDisabled ? 'disabled' : 'default';
  if (_isChecked && chipColor) {
    intentVariant = chipColor;
  }
  const chipBackgroundColor = chipColorTokens.background[intentVariant][stateVariant];
  const chipBorderColor = chipColorTokens.border[intentVariant][stateVariant];

  return (
    <BaseBox
      {...metaAttribute({ name: MetaConstants.Chip, testID })}
      {...getStyledProps(styledProps)}
      display={(isReactNative() ? 'flex' : 'inline-flex') as never}
    >
      <SelectorLabel
        componentName={MetaConstants.ChipLabel}
        onTouchStart={handlePointerPressedIn}
        onTouchEnd={handlePointerPressedOut}
        onMouseDown={handlePointerPressedIn}
        onMouseUp={handlePointerPressedOut}
        onMouseOut={handlePointerPressedOut}
        onKeyDown={handleKeyboardPressedIn}
        onKeyUp={handleKeyboardPressedOut}
        inputProps={isReactNative() ? inputProps : {}}
        style={{ cursor: _isDisabled ? 'not-allowed' : 'pointer' }}
      >
        <BaseBox display="flex" flexDirection="column">
          <BaseBox display="flex" alignItems="center" flexDirection="row">
            <SelectorInput
              hoverTokens={getChipInputHoverTokens(chipColor)}
              isChecked={state.isChecked}
              isDisabled={_isDisabled}
              inputProps={inputProps}
              ref={ref}
            />
            <AnimatedChip
              borderColor={chipBorderColor}
              isDisabled={_isDisabled}
              isPressed={isPressed}
              isDesktop={matchedDeviceType === 'desktop'}
            >
              <StyledChipWrapper
                borderColor={chipBorderColor}
                isChecked={_isChecked}
                isDisabled={_isDisabled}
                color={chipColor}
                display="flex"
                flexDirection="row"
                justifyContent="center"
                alignItems="center"
                overflow="hidden"
                backgroundColor={chipBackgroundColor}
                borderRadius="max"
                borderWidth={['xsmall', 'small'].includes(_size) ? 'thinner' : 'thin'}
                paddingLeft={
                  chipHorizontalPaddingTokens[Boolean(Icon) ? 'withIcon' : 'withoutIcon'].left[
                    _size
                  ]
                }
                paddingRight={
                  chipHorizontalPaddingTokens[Boolean(Icon) ? 'withIcon' : 'withoutIcon'].right[
                    _size
                  ]
                }
                height={makeSize(chipHeightTokens[_size])}
              >
                {Icon ? (
                  <BaseBox paddingRight="spacing.3" display="flex">
                    <Icon color={chipIconColor} size={chipIconSizes[_size]} />
                  </BaseBox>
                ) : null}
                <Text
                  {...chipTextSizes[_size]}
                  type="normal"
                  truncateAfterLines={1}
                  color={chipTextColor}
                >
                  {children}
                </Text>
              </StyledChipWrapper>
            </AnimatedChip>
          </BaseBox>
        </BaseBox>
      </SelectorLabel>
    </BaseBox>
  );
};

const Chip = assignWithoutSideEffects(React.forwardRef(_Chip), {
  displayName: 'Chip',
});

export { Chip };
export type { ChipProps };
