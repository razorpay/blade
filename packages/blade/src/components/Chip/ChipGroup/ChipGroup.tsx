import React from 'react';
import { chipGroupGapTokens } from '../chipTokens';
import { ChipGroupProvider } from './ChipGroupContext';
import { useChipGroup } from './useChipGroup';
import type { ChipGroupProps } from './types';
import BaseBox from '~components/Box/BaseBox';
import { SelectorGroupField } from '~components/Form/Selector/SelectorGroupField';
import { getStyledProps } from '~components/Box/styledProps';
import { isReactNative } from '~utils';
// import { useBreakpoint } from '~utils';

// import { useTheme } from '~components/BladeProvider';
// import { makeSize } from '~utils/makeSize';

const ChipGroup = ({
  children,
  isDisabled = false,
  name,
  defaultValue,
  onChange,
  value,
  size = 'small',
  variant = 'neutral',
  testID,
  selectionType = 'single',
  ...styledProps
}: ChipGroupProps): React.ReactElement => {
  const { contextValue, ids } = useChipGroup({
    defaultValue,
    onChange,
    value,
    isDisabled,
    name,
    size,
    variant,
    selectionType,
  });
  // const { theme } = useTheme();
  // const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
  // const gap = checkboxSizes.group.gap[size][matchedDeviceType];
  // const childCount = React.Children.count(children);

  return (
    <ChipGroupProvider value={contextValue}>
      <BaseBox {...getStyledProps(styledProps)}>
        <SelectorGroupField
          accessibilityRole={isReactNative() && selectionType === 'single' ? 'radiogroup' : 'group'}
          labelledBy={ids.labelId}
          componentName="chip-group"
          testID={testID}
        >
          <BaseBox>
            <BaseBox display="flex" flexDirection="row">
              {React.Children.map(children, (child, index) => {
                return (
                  <BaseBox
                    key={index}
                    marginBottom={chipGroupGapTokens[size].bottom}
                    marginRight={chipGroupGapTokens[size].right}
                  >
                    {child}
                  </BaseBox>
                );
              })}
            </BaseBox>
          </BaseBox>
        </SelectorGroupField>
      </BaseBox>
    </ChipGroupProvider>
  );
};

export { ChipGroup };
export type { ChipGroupProps };
