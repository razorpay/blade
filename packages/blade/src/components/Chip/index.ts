export * from './Chip';
export * from './ChipGroup';
