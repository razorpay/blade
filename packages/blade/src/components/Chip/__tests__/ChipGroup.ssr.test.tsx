import React from 'react';
import { Chip } from '../Chip';
import { ChipGroup } from '../ChipGroup';
import renderWithSSR from '~utils/testing/renderWithSSR.web';
import { Box } from '~components/Box';

describe('<ChipGroup />', () => {
  it('should render with selectionType="single"', () => {
    const { getByRole, container } = renderWithSSR(
      <Box>
        <ChipGroup accessibilityLabel="Select fruits" selectionType="single">
          <Chip value="apple">Apple</Chip>
          <Chip value="mango">Mango</Chip>
          <Chip value="orange">Orange</Chip>
        </ChipGroup>
        <ChipGroup accessibilityLabel="Select flowers" selectionType="multiple">
          <Chip value="Rose">Rose</Chip>
          <Chip value="Sunflower">Sunflower</Chip>
          <Chip value="Lily">Lily</Chip>
        </ChipGroup>
      </Box>,
    );

    expect(getByRole('radiogroup')).toBeInTheDocument();
    expect(getByRole('radio', { name: 'Apple' })).toBeInTheDocument();
    expect(getByRole('radio', { name: 'Mango' })).toBeInTheDocument();

    expect(getByRole('group')).toBeInTheDocument();
    expect(getByRole('checkbox', { name: 'Rose' })).toBeInTheDocument();
    expect(getByRole('checkbox', { name: 'Lily' })).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
