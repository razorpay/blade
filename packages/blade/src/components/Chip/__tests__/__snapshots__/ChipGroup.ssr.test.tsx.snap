// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChipGroup /> should render with selectionType="single" 1`] = `"<div id="root"><div data-blade-component="box" class="BaseBox-bmPWx"><div data-blade-component="base-box" class="BaseBox-bmPWx"><div role="radiogroup" aria-labelledby="chip-group-undefined-label" data-blade-component="chip-group" class="BaseBox-bmPWx djEyfF"><div data-blade-component="base-box" class="BaseBox-bmPWx"><div data-blade-component="visually-hidden" class="VisuallyHiddenweb__StyledVisuallyHidden-g3hls3-0 ijVANw"><p class="StyledBaseText-dVBfTO cxTQmj" data-blade-component="text">Select fruits</p></div><div data-blade-component="base-box" class="BaseBox-bmPWx jjhFBS"><div data-blade-component="base-box" class="BaseBox-bmPWx geZScS"><div data-blade-component="chip" class="BaseBox-bmPWx bcSLaJ"><label style="cursor:pointer" data-blade-component="chip-label" class="SelectorLabelweb__StyledSelectorLabel-sc-1yoi09a-0 bgSmPY"><div data-blade-component="base-box" class="BaseBox-bmPWx xrMLB"><div data-blade-component="base-box" class="BaseBox-bmPWx fSbYvl"><input type="radio" role="radio" aria-required="false" aria-invalid="false" aria-disabled="false" aria-checked="false" aria-describedby="radio-undefined-helptext-undefined" class="SelectorInputweb__StyledInput-sc-1bnkrae-0 jwCAeQ" value="apple"/><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedChipweb__AnimatedChip-sc-1xiynj1-0 dRscRB kDMGeD"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledChipWrapperweb__StyledChipWrapper-sc-1p3u1xx-0 jRMjox iQELOd"><p class="StyledBaseText-dVBfTO fXvaBg" data-blade-component="text">Apple</p></div></div></div></div></label></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx geZScS"><div data-blade-component="chip" class="BaseBox-bmPWx bcSLaJ"><label style="cursor:pointer" data-blade-component="chip-label" class="SelectorLabelweb__StyledSelectorLabel-sc-1yoi09a-0 bgSmPY"><div data-blade-component="base-box" class="BaseBox-bmPWx xrMLB"><div data-blade-component="base-box" class="BaseBox-bmPWx fSbYvl"><input type="radio" role="radio" aria-required="false" aria-invalid="false" aria-disabled="false" aria-checked="false" aria-describedby="radio-undefined-helptext-undefined" class="SelectorInputweb__StyledInput-sc-1bnkrae-0 jwCAeQ" value="mango"/><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedChipweb__AnimatedChip-sc-1xiynj1-0 dRscRB kDMGeD"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledChipWrapperweb__StyledChipWrapper-sc-1p3u1xx-0 jRMjox iQELOd"><p class="StyledBaseText-dVBfTO fXvaBg" data-blade-component="text">Mango</p></div></div></div></div></label></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx geZScS"><div data-blade-component="chip" class="BaseBox-bmPWx bcSLaJ"><label style="cursor:pointer" data-blade-component="chip-label" class="SelectorLabelweb__StyledSelectorLabel-sc-1yoi09a-0 bgSmPY"><div data-blade-component="base-box" class="BaseBox-bmPWx xrMLB"><div data-blade-component="base-box" class="BaseBox-bmPWx fSbYvl"><input type="radio" role="radio" aria-required="false" aria-invalid="false" aria-disabled="false" aria-checked="false" aria-describedby="radio-undefined-helptext-undefined" class="SelectorInputweb__StyledInput-sc-1bnkrae-0 jwCAeQ" value="orange"/><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedChipweb__AnimatedChip-sc-1xiynj1-0 dRscRB kDMGeD"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledChipWrapperweb__StyledChipWrapper-sc-1p3u1xx-0 jRMjox iQELOd"><p class="StyledBaseText-dVBfTO fXvaBg" data-blade-component="text">Orange</p></div></div></div></div></label></div></div></div></div></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx"><div role="group" aria-labelledby="chip-group-undefined-label" data-blade-component="chip-group" class="BaseBox-bmPWx djEyfF"><div data-blade-component="base-box" class="BaseBox-bmPWx"><div data-blade-component="visually-hidden" class="VisuallyHiddenweb__StyledVisuallyHidden-g3hls3-0 ijVANw"><p class="StyledBaseText-dVBfTO cxTQmj" data-blade-component="text">Select flowers</p></div><div data-blade-component="base-box" class="BaseBox-bmPWx jjhFBS"><div data-blade-component="base-box" class="BaseBox-bmPWx geZScS"><div data-blade-component="chip" class="BaseBox-bmPWx bcSLaJ"><label style="cursor:pointer" data-blade-component="chip-label" class="SelectorLabelweb__StyledSelectorLabel-sc-1yoi09a-0 bgSmPY"><div data-blade-component="base-box" class="BaseBox-bmPWx xrMLB"><div data-blade-component="base-box" class="BaseBox-bmPWx fSbYvl"><input type="checkbox" role="checkbox" aria-required="false" aria-invalid="false" aria-disabled="false" aria-checked="false" class="SelectorInputweb__StyledInput-sc-1bnkrae-0 jwCAeQ" value="Rose"/><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedChipweb__AnimatedChip-sc-1xiynj1-0 dRscRB kDMGeD"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledChipWrapperweb__StyledChipWrapper-sc-1p3u1xx-0 jRMjox iQELOd"><p class="StyledBaseText-dVBfTO fXvaBg" data-blade-component="text">Rose</p></div></div></div></div></label></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx geZScS"><div data-blade-component="chip" class="BaseBox-bmPWx bcSLaJ"><label style="cursor:pointer" data-blade-component="chip-label" class="SelectorLabelweb__StyledSelectorLabel-sc-1yoi09a-0 bgSmPY"><div data-blade-component="base-box" class="BaseBox-bmPWx xrMLB"><div data-blade-component="base-box" class="BaseBox-bmPWx fSbYvl"><input type="checkbox" role="checkbox" aria-required="false" aria-invalid="false" aria-disabled="false" aria-checked="false" class="SelectorInputweb__StyledInput-sc-1bnkrae-0 jwCAeQ" value="Sunflower"/><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedChipweb__AnimatedChip-sc-1xiynj1-0 dRscRB kDMGeD"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledChipWrapperweb__StyledChipWrapper-sc-1p3u1xx-0 jRMjox iQELOd"><p class="StyledBaseText-dVBfTO fXvaBg" data-blade-component="text">Sunflower</p></div></div></div></div></label></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx geZScS"><div data-blade-component="chip" class="BaseBox-bmPWx bcSLaJ"><label style="cursor:pointer" data-blade-component="chip-label" class="SelectorLabelweb__StyledSelectorLabel-sc-1yoi09a-0 bgSmPY"><div data-blade-component="base-box" class="BaseBox-bmPWx xrMLB"><div data-blade-component="base-box" class="BaseBox-bmPWx fSbYvl"><input type="checkbox" role="checkbox" aria-required="false" aria-invalid="false" aria-disabled="false" aria-checked="false" class="SelectorInputweb__StyledInput-sc-1bnkrae-0 jwCAeQ" value="Lily"/><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedChipweb__AnimatedChip-sc-1xiynj1-0 dRscRB kDMGeD"><div data-blade-component="base-box" class="BaseBox-bmPWx StyledChipWrapperweb__StyledChipWrapper-sc-1p3u1xx-0 jRMjox iQELOd"><p class="StyledBaseText-dVBfTO fXvaBg" data-blade-component="text">Lily</p></div></div></div></div></label></div></div></div></div></div></div></div></div>"`;

exports[`<ChipGroup /> should render with selectionType="single" 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  margin-bottom: 8px;
  margin-right: 8px;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  border-color: hsla(216,19%,89%,1);
  border-style: solid;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 12px;
  padding-left: 12px;
  height: 28px;
  background-color: transparent;
  border-radius: 9999px;
  border-width: 0.5px;
  border-color: hsla(216,19%,89%,1);
  border-style: solid;
}

.c11 {
  background-color: hsla(0,0%,100%,1);
  border-radius: 9999px;
  border-color: hsla(216,19%,89%,1);
  border-width: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: left;
  text-overflow: ellipsis;
  max-width: 420px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-color: transparent;
}

.c13:hover {
  background-color: hsla(216,15%,54%,0.09);
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 2px;
  margin-bottom: 2px;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: 0 -1px -1px 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  left: -10000px;
  white-space: nowrap;
  word-wrap: normal;
}

.c9 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: 0 -1px -1px 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  left: -10000px;
  white-space: nowrap;
  word-wrap: normal;
}

.c9:focus-visible + div {
  outline: 1px solid hsla(220,30%,96%,1);
  box-shadow: 0px 0px 0px 4px hsla(218,89%,51%,0.18);
}

.c9:hover + div {
  border-color: hsla(214,18%,69%,1);
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  -webkit-transition-duration: 70ms;
  transition-duration: 70ms;
}

.c2 {
  color: hsla(217,56%,17%,1);
  font-family: "Inter",-apple-system,BlinkMacSystemFont,San Francisco,Segoe UI,Roboto,Helvetica Neue,sans-serif;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c14 {
  color: hsla(216,27%,36%,1);
  font-family: "Inter",-apple-system,BlinkMacSystemFont,San Francisco,Segoe UI,Roboto,Helvetica Neue,sans-serif;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
  overflow: hidden;
  display: -webkit-box;
  line-clamp: 1;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

@media screen and (min-width:320px) {
  .c10 {
    border-style: solid;
  }
}

@media screen and (min-width:480px) {
  .c10 {
    border-style: solid;
  }
}

@media screen and (min-width:768px) {
  .c10 {
    border-style: solid;
  }
}

@media screen and (min-width:1024px) {
  .c10 {
    border-style: solid;
  }
}

@media screen and (min-width:1200px) {
  .c10 {
    border-style: solid;
  }
}

@media screen and (min-width:320px) {
  .c12 {
    border-style: solid;
  }
}

@media screen and (min-width:480px) {
  .c12 {
    border-style: solid;
  }
}

@media screen and (min-width:768px) {
  .c12 {
    border-style: solid;
  }
}

@media screen and (min-width:1024px) {
  .c12 {
    border-style: solid;
  }
}

@media screen and (min-width:1200px) {
  .c12 {
    border-style: solid;
  }
}

<div
  id="root"
>
  <div
    class=""
    data-blade-component="box"
  >
    <div
      class=""
      data-blade-component="base-box"
    >
      <div
        aria-labelledby="chip-group-7-label"
        class="c0"
        data-blade-component="chip-group"
        role="radiogroup"
      >
        <div
          class=""
          data-blade-component="base-box"
        >
          <div
            class="c1"
            data-blade-component="visually-hidden"
          >
            <p
              class="c2"
              data-blade-component="text"
            >
              Select fruits
            </p>
          </div>
          <div
            class="c3"
            data-blade-component="base-box"
          >
            <div
              class="c4"
              data-blade-component="base-box"
            >
              <div
                class="c5"
                data-blade-component="chip"
              >
                <label
                  class="c6"
                  data-blade-component="chip-label"
                  style="cursor:pointer"
                >
                  <div
                    class="c7"
                    data-blade-component="base-box"
                  >
                    <div
                      class="c8"
                      data-blade-component="base-box"
                    >
                      <input
                        aria-checked="false"
                        aria-describedby="radio-1-helptext-2"
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        class="c9"
                        role="radio"
                        type="radio"
                        value="apple"
                      />
                      <div
                        class="c10 c11"
                        data-blade-component="base-box"
                      >
                        <div
                          class="c12 c13"
                          data-blade-component="base-box"
                        >
                          <p
                            class="c14"
                            data-blade-component="text"
                          >
                            Apple
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </div>
            </div>
            <div
              class="c4"
              data-blade-component="base-box"
            >
              <div
                class="c5"
                data-blade-component="chip"
              >
                <label
                  class="c6"
                  data-blade-component="chip-label"
                  style="cursor:pointer"
                >
                  <div
                    class="c7"
                    data-blade-component="base-box"
                  >
                    <div
                      class="c8"
                      data-blade-component="base-box"
                    >
                      <input
                        aria-checked="false"
                        aria-describedby="radio-3-helptext-4"
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        class="c9"
                        role="radio"
                        type="radio"
                        value="mango"
                      />
                      <div
                        class="c10 c11"
                        data-blade-component="base-box"
                      >
                        <div
                          class="c12 c13"
                          data-blade-component="base-box"
                        >
                          <p
                            class="c14"
                            data-blade-component="text"
                          >
                            Mango
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </div>
            </div>
            <div
              class="c4"
              data-blade-component="base-box"
            >
              <div
                class="c5"
                data-blade-component="chip"
              >
                <label
                  class="c6"
                  data-blade-component="chip-label"
                  style="cursor:pointer"
                >
                  <div
                    class="c7"
                    data-blade-component="base-box"
                  >
                    <div
                      class="c8"
                      data-blade-component="base-box"
                    >
                      <input
                        aria-checked="false"
                        aria-describedby="radio-5-helptext-6"
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        class="c9"
                        role="radio"
                        type="radio"
                        value="orange"
                      />
                      <div
                        class="c10 c11"
                        data-blade-component="base-box"
                      >
                        <div
                          class="c12 c13"
                          data-blade-component="base-box"
                        >
                          <p
                            class="c14"
                            data-blade-component="text"
                          >
                            Orange
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class=""
      data-blade-component="base-box"
    >
      <div
        aria-labelledby="chip-group-26-label"
        class="c0"
        data-blade-component="chip-group"
        role="group"
      >
        <div
          class=""
          data-blade-component="base-box"
        >
          <div
            class="c1"
            data-blade-component="visually-hidden"
          >
            <p
              class="c2"
              data-blade-component="text"
            >
              Select flowers
            </p>
          </div>
          <div
            class="c3"
            data-blade-component="base-box"
          >
            <div
              class="c4"
              data-blade-component="base-box"
            >
              <div
                class="c5"
                data-blade-component="chip"
              >
                <label
                  class="c6"
                  data-blade-component="chip-label"
                  style="cursor:pointer"
                >
                  <div
                    class="c7"
                    data-blade-component="base-box"
                  >
                    <div
                      class="c8"
                      data-blade-component="base-box"
                    >
                      <input
                        aria-checked="false"
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        class="c9"
                        role="checkbox"
                        type="checkbox"
                        value="Rose"
                      />
                      <div
                        class="c10 c11"
                        data-blade-component="base-box"
                      >
                        <div
                          class="c12 c13"
                          data-blade-component="base-box"
                        >
                          <p
                            class="c14"
                            data-blade-component="text"
                          >
                            Rose
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </div>
            </div>
            <div
              class="c4"
              data-blade-component="base-box"
            >
              <div
                class="c5"
                data-blade-component="chip"
              >
                <label
                  class="c6"
                  data-blade-component="chip-label"
                  style="cursor:pointer"
                >
                  <div
                    class="c7"
                    data-blade-component="base-box"
                  >
                    <div
                      class="c8"
                      data-blade-component="base-box"
                    >
                      <input
                        aria-checked="false"
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        class="c9"
                        role="checkbox"
                        type="checkbox"
                        value="Sunflower"
                      />
                      <div
                        class="c10 c11"
                        data-blade-component="base-box"
                      >
                        <div
                          class="c12 c13"
                          data-blade-component="base-box"
                        >
                          <p
                            class="c14"
                            data-blade-component="text"
                          >
                            Sunflower
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </div>
            </div>
            <div
              class="c4"
              data-blade-component="base-box"
            >
              <div
                class="c5"
                data-blade-component="chip"
              >
                <label
                  class="c6"
                  data-blade-component="chip-label"
                  style="cursor:pointer"
                >
                  <div
                    class="c7"
                    data-blade-component="base-box"
                  >
                    <div
                      class="c8"
                      data-blade-component="base-box"
                    >
                      <input
                        aria-checked="false"
                        aria-disabled="false"
                        aria-invalid="false"
                        aria-required="false"
                        class="c9"
                        role="checkbox"
                        type="checkbox"
                        value="Lily"
                      />
                      <div
                        class="c10 c11"
                        data-blade-component="base-box"
                      >
                        <div
                          class="c12 c13"
                          data-blade-component="base-box"
                        >
                          <p
                            class="c14"
                            data-blade-component="text"
                          >
                            Lily
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
